src/bin/pg_autoctl/cli_do_monitor.c

```diff
@@ -223,17 +223,16 @@ cli_do_monitor_get_other_nodes(int argc, char **argv)
 	{
 		char json[BUFSIZE];
 
-		if (!monitor_get_other_nodes_as_json(&monitor,
-											 config.nodename,
-											 config.pgSetup.pgport,
-											 ANY_STATE,
-											 json, BUFSIZE))
+		if (!monitor_print_other_nodes_as_json(&monitor,
```

Comment:
I did not like the idea of passing stdout as file stream.

Can  we switch to old way but instead use a dynamic buffer allocated inside ?  We already allocate buffer in `parseSingleValueResult` function which we did not deallocate. We can just pass/return a pointer to that buffer, print it inside this (`cli_do_monitor_get_other_nodes`) function like we did before and free the buffer afterwards .

Reply:
I can understand “I don't like” as a discussion opener, but it's a little short as the whole reason for refusing to approve a PR. Can you please expand on why you do not like using a `FILE *stream` argument to a function, when all the function now does is print out a result on said stream?

Reply:
agree that "I don't like" was not a good start here.   my reasoning is that, the function is always called with the same parameter that is `stdout` we don't open or create that file descriptor or accept as parameter to calling function.

Since the function name contains `...print...`, it is explicit that we will print something somewhere, if we have to do printing inside the function I would prefer use of stdout is inside.

In the second part of the function. We are separating  data fetching and printing duties. I was thinking about using the same approach in both with and without --json. Perhaps we can merge fetch with print functions and just have 2 function calls for with/without --json.

Reply:
I can see now that the API could be improved, because we have `print_as_json` functions and could expect similar API for the tabular format. I have done that in the revised version of the PR, thanks!